src/riscv/lib/src/state_backend/normal_backend.rs

```diff
@@ -179,25 +179,24 @@ pub(crate) mod test_helpers {
 
     /// Ensure [`Cells`] can be serialised and deserialised in a consistent way.
     #[test]
-    fn cells_serialise() {
+    fn atoms_serialise() {
         proptest::proptest!(|(a: u64, b: u64, c: u64)|{
-            let cell: Cells<u64, 3, Normal> = Cells::bind([a, b, c]);
+            let cell: Atom<_, Normal> = Atom::new([a, b, c]);
             let bytes = serialise(&cell).unwrap();
 
-            let cell_after: Cells<u64, 3, Normal> = deserialise(&bytes).unwrap();
+            let cell_after: Atom<[u64; 3], Normal> = deserialise(&bytes).unwrap();
 
-            assert_eq!(cell.read_all(), cell_after.read_all());
+            assert_eq!(cell.read(), cell_after.read());
 
             for i in 0..3 {
-                assert_eq!(cell.read(i), cell_after.read(i));
+                assert_eq!(cell[i], cell_after[i]);
             }
 
             let bytes_after = serialise(&cell_after).unwrap();
             assert_eq!(bytes, bytes_after);
 
             // Serialisation is consistent with that of the `Prove` mode.
-            let proof_cells: Cells<u64, 3, Prove> =
-                Cells::bind(ProofRegion::bind(cell.region_ref()));
+            let proof_cells: Atom<_, Prove> = cell.start_proof();
```

Comment:
nit - `s/cell/atom/`?

throughout the test we now have things called cell/cells which are now atoms 😄